### PR TITLE
Additional widgets can be added to the livreGUI

### PR DESCRIPTION
**Pull request by bilgili**

### CODE REVIEW

**Comment by bilgili** [comment]
retest this please

**Comment by bilgili** [comment]
retest this please

**Comment by favreau** [comment]
retest this please

**Comment by favreau** [comment]
Documentation of header files would ease the review process

**Comment by bilgili** [comment]
Header files have documentation necessary ?

**Comment by favreau** [comment]
Not in PropertyEditDelegate.h and GeneratorPropertiesModel.h (I started by those ones... other headers look ok)

File: apps/livreGUI/Controller.cpp [hunk 1]
```diff
@@ -21,45 +21,159 @@
...
+        _subscriberPoll.join();
+    }
+
+    bool connect( const std::string&, const uint16_t )
+    {
+        LBUNIMPLEMENTED;
+        return false;
+    }
+
+    std::string getURIStr( const servus::URI& uri )
```
Comment (chevtche) [inline apps/livreGUI/Controller.cpp]: I think this method is const??

**Comment by bilgili** [comment]
updated

File: apps/livreGUI/Controller.cpp [hunk 2]
```diff
@@ -21,45 +21,159 @@
...
+    }
+
+    template< typename Type >
+    Type* getObject( std::map< std::string, Type* >& map,
+                const servus::URI& uri  )
+    {
+        const std::string& uriStr = getURIStr( uri );
+        typename std::map< std::string, Type* >::iterator it = map.find( uriStr );
+        if( it == map.end( ))
+            map[ uriStr ] = new Type( uri );
```
Comment (favreau) [inline apps/livreGUI/Controller.cpp]: No corresponding delete? ~impl should properly destroy allocated publishers and subscribers.

**Comment by bilgili** [comment]
retest this please

**Comment by bilgili** [comment]
retest this please

**Comment by tribal-tec** [comment]
Needs this PR first, sorry: https://github.com/BlueBrain/Livre/pull/116

Or you merge before, and I fix the build afterwards.

**Comment by bilgili** [comment]
please retest this

**Comment by chevtche** [comment]
retest this please

**Comment by bilgili** [comment]
retest this please

**Comment by bilgili** [comment]
retest this please

**Comment by bilgili** [comment]
retest this please

**Comment by bilgili** [comment]
retest this please

File: .gitsubprojects [hunk 3]
```diff
@@ -5,3 +5,4 @@ git_subproject(dash https://github.com/BlueBrain/dash.git f43fd21)
 git_subproject(zeq https://github.com/HBPVis/zeq.git acaadb1)
 git_subproject(Collage https://github.com/Eyescale/Collage.git 182698e)
 git_subproject(Equalizer https://github.com/Eyescale/Equalizer.git 0beb1bc)
+git_subproject(ISC ssh://bbpcode.epfl.ch/viz/ISC.git a3fd2f5)
```
Comment (tribal-tec) [inline .gitsubprojects]: This breaks the build for everybody outside of BBP.

**Pull request merged**